src/user/use-cases/sign-in/sign-in.spec.ts

```diff
@@ -26,88 +28,79 @@ describe('Sign In Use Case', () => {
           },
         },
         {
-          provide: JwtService,
+          provide: JwtAuthProvider,
           useValue: {
-            verifyAsync: jest.fn(),
+            decodeToken: jest.fn(),
           },
         },
       ],
     }).compile();
 
     signInUseCase = module.get<SignInUseCase>(SignInUseCase);
     usersRepository = module.get<IUserRepository>(IUserRepository);
-    jwtService = module.get<JwtService>(JwtService);
+    jwtAuthProvider = module.get<JwtAuthProvider>(JwtAuthProvider);
 
     userPayload = {
       sub: 'auth0|58vfb567d5asdea52bc65ebba',
       name: 'John Doe',
       email: 'john.doe@example.com',
     };
+
+    findByEmailMock = jest.spyOn(usersRepository, 'findByEmail');
+    createUserMock = jest.spyOn(usersRepository, 'create');
+    decodeTokenMock = jest.spyOn(jwtAuthProvider, 'decodeToken');
   });
 
   beforeEach(() => {
-    jest.spyOn(jwtService, 'verifyAsync').mockResolvedValue(userPayload);
+    decodeTokenMock.mockReturnValue(userPayload);
   });
 
-  const mockFindByEmail = (user: User | null) => {
-    jest.spyOn(usersRepository, 'findByEmail').mockResolvedValue(user);
-  };
-
-  const mockCreateUser = (user: SaveUserDTO) => {
-    jest.spyOn(usersRepository, 'create').mockResolvedValue(user as User);
-  };
-
   const token = 'valid-jwt-token';
   const invalidToken = 'invalid-jwt-token';
 
-  it('should be able to sign in an user with a valid token', async () => {
-    const expectedUser: User = {
-      id: 'auth0|58vfb567d5asdea52bc65ebba',
-      name: 'John Doe',
-      email: 'john.doe@example.com',
-    } as User;
+  it('should return 200 if user already exists', async () => {
```

Comment:
Isso é detalhe, mas vocês podem ser mais descritivos quando escreverem os its.
Ex: should be able to return a success response if user already exists.
Por que da forma como ta escrito aqui parece que o método deve retornar o número 200.